apps/server/src/modules/tldraw/controller/tldraw-ws.controller.ts

```diff
@@ -0,0 +1,56 @@
+import { WebSocketGateway, WebSocketServer, OnGatewayInit, OnGatewayConnection } from '@nestjs/websockets';
+import { Server, WebSocket } from 'ws';
+import { MongodbPersistence } from 'y-mongodb-provider';
+import { ConfigService } from '@nestjs/config';
+import { TldrawConfig, SOCKET_PORT } from '@src/modules/tldraw/config';
+import { WsCloseCodeEnum } from '@src/modules/tldraw/types/ws-close-code-enum';
+import { setupWSConnection, setPersistence, updateDocument } from '../utils';
+
+@WebSocketGateway(SOCKET_PORT)
+export class TldrawWsController implements OnGatewayInit, OnGatewayConnection {
+	@WebSocketServer()
+	server!: Server;
+
+	connectionString: string;
+
+	constructor(readonly configService: ConfigService<TldrawConfig, true>) {
+		this.connectionString = this.configService.get<string>('CONNECTION_STRING');
+	}
+
+	handleConnection(client: WebSocket, request: Request) {
+		const docName = this.getDocNameFromRequest(request);
+
+		if (docName.length > 0 && this.configService.get<string>('FEATURE_TLDRAW_ENABLED')) {
+			setupWSConnection(client, docName);
+		} else {
+			client.close(
+				WsCloseCodeEnum.WS_CUSTOM_CLIENT_CLOSE_CODE,
+				'Document name is mandatory in url or Tldraw Tool is turned off.'
+			);
+		}
+	}
+
+	afterInit() {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment,@typescript-eslint/no-unsafe-call
+		const mdb = new MongodbPersistence(this.connectionString, {
```

Comment:
The persistence should be a detail and unknown in controller layer. This is part of the repo layer.
We have currently following layer: MongoDB/external service --|--> Repo -> Domain[ Service <- UC] <- controller.